blog/tests.py

```diff
@@ -118,3 +139,223 @@ def test_delete_blog(self):
         force_authenticate(request, user=user)
         response = view(request, slug=blog.slug)
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_update_blog(self):
+        """
+        Tests to update a Blog Post Successfully.
+        """
+        body = {
+            'title': 'Test',
+            'description': 'This is a Test Blog.'
+        }
+        request = self.create_blog_request(body)
+        user = User.objects.get(username='test')
+        view = BlogAPI.as_view({
+            'post': 'create'
+        })
+
+        force_authenticate(request, user=user)
+        response = view(request)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        blog = Blog.objects.get()
+
+        body = {
+            'title': 'Test101',
+            'description': 'This is a Test Blog. Amirite! :D'
+        }
+
+        factory = APIRequestFactory()
+        url = '/api/blog/'
+
+        request = factory.put(url, body)
+        view = BlogAPI.as_view({
+            'put': 'update'
+        })
+
+        force_authenticate(request, user=user)
+        response = view(request, slug=blog.slug)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(Blog.objects.count(), 1)
+        self.assertEqual(Blog.objects.get().title, 'Test101')
+
+    def test_search_blog_list(self):
+        """
+        Tests to search Blog Posts Successfully.
+        """
+        body = {
+            'title': 'Test',
+            'description': 'This is a Test Blog.'
+        }
+        request = self.create_blog_request(body)
+        user = User.objects.get(username='test')
+        view = BlogAPI.as_view({
+            'post': 'create'
+        })
+
+        force_authenticate(request, user=user)
+        response = view(request)
+
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
Make a helper method to create a blog post. As in this test and the test above we require a blog post before verifying that it updates or searched correctly.

Reply:
Done.